### ISSUE TITLE
PostgREST over CockroachDB support

### ISSUE THREAD

**Issue by cmnstmntmn** [PostgREST/postgrest#1140]
can PostgREST be user over CockroachDB instead of PostgreSQL ?

**Comment by tmikaeld** [comment]
@steve-chavez Is this still the case? Scaling writes toward PostgreSQL is still hard and something that CockroachDB handles with excellence.

**Comment by steve-chavez** [comment]
@tmikaeld Yes, `set local` is still unsupported. See https://github.com/cockroachdb/cockroach/issues/32562

**Comment by nicksrandall** [comment]
`set local` is now supported (albeit not in a transaction). Anything that CockroachDB is missing before it can be reasonably used by PostgREST?

**Comment by wolfgangwalther** [comment]
Well first of all, exactly that: `set local` in a transaction. I'm not sure about the [`pg_catalog` tables,](https://www.cockroachlabs.com/docs/stable/pg-catalog.html#data-exposed-by-pg_catalog) which we feed our schema cache from:

> The tables in the pg_catalog schema roughly correspond to the system catalogs in PostgreSQL.

That would need to be tested, probably.

`json_to_recordset` is used but not supported, yet: https://github.com/cockroachdb/cockroach/issues/70037

(this is not a full list)

**Comment by steve-chavez** [comment]
Cockroach has actually been trying to give full support for PostgREST, some features do work: https://github.com/cockroachdb/cockroach/issues/69010#issuecomment-921440081.

**Comment by steve-chavez** [comment]
> json_to_recordset is used but not supported, yet: https://github.com/cockroachdb/cockroach/issues/70037

Actually not that relevant since it's only used for RPC and cockroachdb doesn't support user-defined functions IIRC.

**Comment by jordanlewis** [comment]
FYI, CockroachDB has completed all of the issues on https://github.com/cockroachdb/cockroach/issues/70037.

**Comment by jordanlewis** [comment]
Err, I meant https://github.com/cockroachdb/cockroach/issues/69010.

**Comment by nicksterious** [comment]
Anyone tried it?

**Comment by ranok** [comment]
@nicksterious I have tried to follow the tutorial on the PostGREST site with a CDB cloud instance. I am able to perform GET verbs, but any others return a 404.

**Comment by mcadecio** [comment]
@nicksterious / @ranok / @nemesisqp  I wrote this in order to have a REST API for any Cockroach database. Its used extensively where I work.
https://github.com/mcadecio/universal-database-proxy-rest 
I would love to get some feedback on it.
PS: It also works with PostgresDB
